src/vos/vos_layout.h

```diff
@@ -175,96 +173,96 @@ enum vos_dtx_record_types {
-	DTX_EF_LEADER			= (1 << 1),
-};
+#define DTX_INLINE_REC_CNT	4
+#define DTX_REC_CAP_DEFAULT	4
 
-/**
- * Persisted DTX entry, it is referenced by btr_record::rec_off
- * of btree VOS_BTR_DTX_TABLE.
- */
-struct vos_dtx_entry_df {
+
+/** Active DTX entry on-disk layout in both SCM and DRAM. */
+struct vos_dtx_act_ent_df {
 	/** The DTX identifier. */
-	struct dtx_id			te_xid;
+	struct dtx_id			dae_xid;
 	/** The identifier of the modified object (shard). */
-	daos_unit_oid_t			te_oid;
+	daos_unit_oid_t			dae_oid;
 	/** The hashed dkey if applicable. */
-	uint64_t			te_dkey_hash;
+	uint64_t			dae_dkey_hash;
 	/** The epoch# for the DTX. */
-	daos_epoch_t			te_epoch;
-	/** Pool map version. */
-	uint32_t			te_ver;
-	/** DTX status, see enum dtx_status. */
-	uint32_t			te_state;
-	/** DTX flags, see enum vos_dtx_entry_flags. */
-	uint32_t			te_flags;
+	daos_epoch_t			dae_epoch;
+	/** The server generation when handles the DTX. */
+	uint64_t			dae_srv_gen;
+	/** The active DTX entry on-disk layout generation. */
+	uint64_t			dae_layout_gen;
 	/** The intent of related modification. */
-	uint32_t			te_intent;
-	/** The timestamp when handles the transaction. */
-	uint64_t			te_time;
-	/** The list of vos_dtx_record_df in SCM. */
-	umem_off_t			te_records;
-	/** The next committed DTX in global list. */
-	umem_off_t			te_next;
-	/** The prev committed DTX in global list. */
-	umem_off_t			te_prev;
+	uint32_t			dae_intent;
+	/** The index in the current vos_dtx_scm_blob. */
+	uint32_t			dae_index;
+	/** The inlined dtx records. */
+	struct vos_dtx_record_df	dae_rec_inline[DTX_INLINE_REC_CNT];
+	/** DTX flags, see enum vos_dtx_entry_flags. */
+	uint32_t			dae_flags;
+	/** The DTX records count, including inline case. */
+	uint32_t			dae_rec_cnt;
+	/** The offset for the list of vos_dtx_record_df if out of inline. */
+	umem_off_t			dae_rec_off;
 };
 
-/**
- * DAOS two-phase commit transaction table.
- */
-struct vos_dtx_table_df {
-	/** The count of committed DTXs in the table. */
-	uint64_t			tt_count;
-	/** The list head of committed DTXs. */
-	umem_off_t			tt_entry_head;
-	/** The list tail of committed DTXs. */
-	umem_off_t			tt_entry_tail;
-	/** The root of the B+ tree for committed DTXs. */
-	struct btr_root			tt_committed_btr;
-	/** The root of the B+ tree for active (prepared) DTXs. */
-	struct btr_root			tt_active_btr;
+/* Assume dae_rec_cnt is next to dae_flags. */
+D_CASSERT(offsetof(struct vos_dtx_act_ent_df, dae_rec_cnt) ==
+	  offsetof(struct vos_dtx_act_ent_df, dae_flags) +
+	  sizeof(((struct vos_dtx_act_ent_df *)0)->dae_flags));
+
+/* Assume dae_rec_off is next to dae_rec_cnt. */
+D_CASSERT(offsetof(struct vos_dtx_act_ent_df, dae_rec_off) ==
+	  offsetof(struct vos_dtx_act_ent_df, dae_rec_cnt) +
+	  sizeof(((struct vos_dtx_act_ent_df *)0)->dae_rec_cnt));
+
+/** Committed DTX entry on-disk layout in both SCM and DRAM. */
+struct vos_dtx_cmt_ent_df {
+	struct dtx_id			dce_xid;
+	daos_epoch_t			dce_epoch;
 };
 
+struct vos_dtx_scm_blob {
```

Comment:
shouldn't it has _df as postfix and remove _scm?

Reply:
OK